src/utils/hooks.ts

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { useCallback, useContext, useEffect, useMemo } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { OnboardingContext } from './OnboardingProvider';
+
+export const useGuardedOnboarding = () => {
+  const onboarding = useContext(OnboardingContext);
```

Comment:
should we be more clear and distinguish the variable names for `useContext(OnboardingContext)` and `useGuardedOnboarding()`? I noticed we set both of these equal to `onboarding`

Reply:
I don't think the naming would change since the content that is returned by `useGuardedOnboarding` is the same as `useContext(OnboardingContext)` - both returns the onboarding context.

The only difference is that `useGuardedOnboarding` adds additional checks, which reduces repeated code